### PR TITLE
Single publisher design update

### PR DESCRIPTION
**Pull request by matthewclegg**
This PR proposes changes to the experimental design for single publisher models based on learnings from the colab (PR 63).

### CODE REVIEW

File: src/driver/experimental_trial.py [hunk 1]
```diff
@@ -195,6 +197,11 @@ def _make_independent_vars_dataframe(self) -> pd.DataFrame:
                 "max_frequency": [
                     self._trial_descriptor.experiment_params.max_frequency
                 ],
+                "average_spend_fraction": [
```
Comment (jiayu-google) [inline src/driver/experimental_trial.py:200]: This is great to add.  Thanks!

File: src/data_generators/single_publisher_design.py [hunk 2]
```diff
@@ -33,34 +33,96 @@
...
         {"gamma_shape": 1.0, "gamma_scale": 4.0},
     ),
-    GeneratorParameters("HeavyTailed", HeavyTailedImpressionGenerator, {"zeta_s": 2.1}),
+    # Gamma Poisson: Mean = 5, Var = 40
+    GeneratorParameters(
+        "Heterogeneous",
+        HeterogeneousImpressionGenerator,
+        {"gamma_shape": 0.44444, "gamma_scale": 9.0},
+    ),
+    # Zeta: Mean = 5, Var undefined
```
Comment (jiayu-google) [inline src/data_generators/single_publisher_design.py:101]: Perhaps clarify that "var = infinity"   (as the series does not converge.)

File: src/data_generators/single_publisher_design.py [hunk 3]
```diff
@@ -33,34 +33,96 @@
...
+        HeterogeneousImpressionGenerator,
+        {"gamma_shape": 1.0, "gamma_scale": 9.0},
+    ),
+    # Gamma Poisson: Mean = 10, Var = 200
+    GeneratorParameters(
+        "Heterogeneous",
+        HeterogeneousImpressionGenerator,
+        {"gamma_shape": 0.42408, "gamma_scale": 21.2222},
+    ),
+    # Zeta: Mean = 10, Var undefined
```
Comment (jiayu-google) [inline src/data_generators/single_publisher_design.py:122]: Ditto

File: src/data_generators/single_publisher_design.py [hunk 4]
```diff
@@ -33,34 +33,96 @@
...
+        "Homogeneous", HomogeneousImpressionGenerator, {"poisson_lambda": 0.5}
     ),
+    # Exponential Poisson: Mean = 1.5, Var = 0.75
     GeneratorParameters(
-        "Homogeneous", HomogeneousImpressionGenerator, {"poisson_lambda": 5.0}
+        "Heterogeneous",
+        HeterogeneousImpressionGenerator,
+        {"gamma_shape": 1.0, "gamma_scale": 0.5},
     ),
+    # Gamma Poisson: Mean = 1.5, Var = 6
```
Comment (jiayu-google) [inline src/data_generators/single_publisher_design.py:53]: Nice!  IIUC, you calculated variance using the conditional variance formula:  Var(freq) = E[Var(X | lambda)] + Var[ E(X | lambda) ] = E(lambda) + Var(lambda) = gamma_shape * gamma_scale + gamma_shape * gamma_scale^2,

while E(freq) = 1 + gamma_shape * gamma_scale.  Is that correct?  Please call out the formula to help readers understand.

**Review by jiayu-google — APPROVED** [review]
LGTM!

File: src/data_generators/single_publisher_design.py (hunk 4 above)
Comment (jiayu-google) [reply src/data_generators/single_publisher_design.py:53]: Perhaps just clarify that it is equivalent to a negative binomial and variance is obtained using negative binomial

File: src/data_generators/single_publisher_design.py (hunk 2 above)
Comment (matthewclegg) [reply src/data_generators/single_publisher_design.py:101]: Done

File: src/data_generators/single_publisher_design.py (hunk 3 above)
Comment (matthewclegg) [reply src/data_generators/single_publisher_design.py:122]: Done

File: src/data_generators/single_publisher_design.py (hunk 4 above)
Comment (matthewclegg) [reply src/data_generators/single_publisher_design.py:53]: Done.

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**